### ISSUE TITLE
Support getting/setting properties by name

### ISSUE THREAD

**Issue by ctrimble** [joelittlejohn/jsonschema2pojo#382]
As the types I am working with evolve, new properties are being introduced which leads to code like this:

```
object.getAdditionalProperties().get("newProperty");
```

Unfortunately, this code will now break if `newProperty` is actually defined in the schema, since the values will start ending up in the object field, instead of the additionalProperties map.  I am wondering some methods like this would help this situation:

```
object.get$("newProperty")
object.set$("newProperty", "value")
```

Where the implementation would first look for the property in a field and then go to the additional properties map.

@joelittlejohn thoughts?

**Comment by joelittlejohn** [comment]
That actually looks pretty awesome. The additionalProperties support produces some quite long-winded methods and I think a lot of people would appreciate being able to interact with them more neatly. I had never actually considered the problem you outlined here (where an additional property later gets 'formalized' in the schema) and the fact that it fixes this is a bonus.

I guess my only concern here would be performance. With a naive implementation it would be quite costly to use these methods, and the more that we encouraged the use of this (for better compatibility with future changes) the slower working with the types would become. It would be good to index the available getters/setters inside `get$` and `set$` to make sure they're as fast as possible.

**Comment by ctrimble** [comment]
Was thinking of implementing with a string switch statement, with all of the known keys generated into the switch.  Seems like that would produce OK performance.  Definitely no reflection.

I will work up a PR and submit it.  If you have any more ideas, please let me know.

**Comment by ctrimble** [comment]
The project is currently set to Java 1.6, but Travis CI testing targets Java 1.7 and Java 1.8.  If I pull the project up to 1.7 (so string switches can be used), is that going to be an issue?

**Comment by joelittlejohn** [comment]
I don't think this is a problem. I'll probably jump to 0.5 if you do this.

**Comment by ctrimble** [comment]
implemented in #386
